src/main/java/io/lettuce/core/api/async/RedisJsonAsyncCommands.java

```diff
@@ -0,0 +1,300 @@
+     * is replaced with the JSON value. For existing keys, when the path exists, except for the last element, a new child is
+     * added with the JSON value.
+     * <p>
+     * Adds a key (with its respective value) to a JSON Object (in a RedisJSON data type key) only if it is the last child in
+     * the path, or it is the parent of a new child being added in the path. Optional arguments NX and XX modify this behavior
+     * for both new RedisJSON data type keys and the JSON Object keys in them.
+     *
+     * @param key the key holding the JSON document.
+     * @param jsonPath the {@link JsonPath} pointing to the value(s) where we want to set the value.
+     * @param value the {@link JsonValue} to set.
+     * @param options the {@link JsonSetArgs} the options for setting the value.
+     * @return String "OK" if the set was successful, null if the {@link JsonSetArgs} conditions are not met.
+     * @since 6.5
+     */
+    RedisFuture<String> jsonSet(K key, JsonPath jsonPath, JsonValue<K, V> value, JsonSetArgs options);
+
+    /**
+     * Append the json-string values to the string at the provided {@link JsonPath} in the JSON document.
+     *
+     * @param key the key holding the JSON document.
+     * @param jsonPath the {@link JsonPath} pointing to the value(s) where we want to append the value.
+     * @param value the {@link JsonValue} to append.
+     * @return Long the new length of the string, or null if the matching JSON value is not a string.
+     * @since 6.5
+     */
+    RedisFuture<List<Long>> jsonStrappend(K key, JsonPath jsonPath, JsonValue<K, V> value);
+
+    /**
+     * Report the length of the JSON String at the provided {@link JsonPath} in the JSON document.
+     *
+     * @param key the key holding the JSON document.
+     * @param jsonPath the {@link JsonPath} pointing to the value(s).
+     * @return Long (in recursive descent) the length of the JSON String at the provided {@link JsonPath}, or null if the value
+     *         ath the desired path is not a string.
+     * @since 6.5
+     */
+    RedisFuture<List<Long>> jsonStrlen(K key, JsonPath jsonPath);
+
+    /**
+     * Toggle a Boolean value stored at the provided {@link JsonPath} in the JSON document.
+     *
+     * @param key the key holding the JSON document.
+     * @param jsonPath the {@link JsonPath} pointing to the value(s).
+     * @return List<Long> the new value after the toggle, 0 for false, 1 for true or null if the path does not exist.
+     * @since 6.5
+     */
+    RedisFuture<List<Long>> jsonToggle(K key, JsonPath jsonPath);
+
+    /**
+     * Report the type of JSON value at the provided {@link JsonPath} in the JSON document.
+     *
+     * @param key the key holding the JSON document.
+     * @param jsonPath the {@link JsonPath} pointing to the value(s).
+     * @return List<JsonType> the type of JSON value at the provided {@link JsonPath}
+     * @since 6.5
+     */
+    RedisFuture<List<V>> jsonType(K key, JsonPath jsonPath);
```

Comment:
I hope its ok to comment here. 
Isn't this guaranteed to be a List<String> per https://redis.io/docs/latest/commands/json.type/ ?
Also what is this type JsonType thats mentioned in the comments , I dont see it anywhere

Reply:
Hey @yisraelU 
Sure, this is an open PR, any comments are welcome.

The JsonType is a leftover from some of the initial API changes I did, I was planning to have a sort of enumeration that would allow for dynamically introspecting the structure of the JSON document.

Right now the implementation would return a String that the app using the driver needs to interpret itself.

Do you favour the first approach more? If so can you give me some use cases to consider?

Reply:
Hey @tishun , thanks for inviting feedback. And once again thanks for an amazing open source project.
I was indeed expecting a List<string> . 
I think in general the question begs what is the value in this command, and assuming the answer is to be able to correctly perform another command that is dependent on the type information , there would be value in having an enumeration instead of a string , as it will reduce bugs around pattern matching , in particular in later versions of Java

Reply:
To clarify I meant `List<String>` and `List<JsonType>` as the two options that were considered.

>  it will reduce bugs around pattern matching , in particular in later versions of Java

Agreed, will add this to the list of leftovers